src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/service/ReferralService.kt

```diff
@@ -1,11 +1,31 @@
 package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.service
 
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.DraftReferral
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ReferralRepository
+import java.util.UUID
 
-object ReferralService {
-  fun updateReferral(original: Referral, update: DraftReferral): Referral {
-    update.completionDeadline?.let { original.completionDeadline = it }
-    return original
+@Service
+class ReferralService(val repository: ReferralRepository) {
+
+  fun createReferral(): Referral {
+    return repository.save(Referral())
+  }
+
+  fun getReferral(id: UUID): Referral? {
+    return repository.findByIdOrNull(id)
+  }
+
+  fun updateReferral(id: UUID, update: DraftReferral): Referral? {
+    val ref = repository.findByIdOrNull(id)
+    if (ref != null) {
```

Comment:
nitpick:

in general i prefer 'guard clauses' to avoid ever expanding levels of indentation

i.e. `if (ref == null) return null`

however kotlin has a handy helper for this type of situation. this can be written as:

```
val ref = getReferral(id) ?: return null 

...
```

Reply:
I'm happy either way, updated